.travis.yml

```diff
@@ -13,56 +13,83 @@ notifications:
 cache:
   directories:
     - $HOME/.composer/cache
-    - $HOME/.phpbrew
+    - $HOME/.jest-cache
     - $HOME/.npm
-
-before_install:
-  - nvm install && nvm use
-  - npm install npm -g
+    - $HOME/.nvm/.cache
+    - $HOME/.phpbrew
 
 branches:
   only:
     - master
 
+before_install:
+  - nvm install
+
 jobs:
   include:
-    - stage: test
+    - name: JS unit tests
       env: WP_VERSION=latest
+      before_install:
+        - nvm install --latest-npm
+      install:
+        - npm ci
       script:
-        - npm install || exit 1
-        - npm run ci || exit 1
+        - npm run lint
+        - npm run check-local-changes
+        - npm run test-unit -- --ci --maxWorkers=2 --cacheDirectory="$HOME/.jest-cache"
 
-    - stage: test
+    - name: PHP unit tests (Docker)
       env: WP_VERSION=latest DOCKER=true
       script:
         - ./bin/run-wp-unit-tests.sh
 
-    - stage: test
+    - name: PHP unit tests (PHP 5.6)
+      language: php
       php: 5.6
       env: WP_VERSION=latest
       script:
         - ./bin/run-wp-unit-tests.sh
       if: branch = master and type != "pull_request"
 
-    - stage: test
-      php: 7.1
+    - name: PHP unit tests (PHP 5.3)
       env: WP_VERSION=latest SWITCH_TO_PHP=5.3
       script:
         - ./bin/run-wp-unit-tests.sh
       if: branch = master and type != "pull_request"
 
-    - stage: test
-      php: 7.1
+    - name: PHP unit tests (PHP 5.2)
       env: WP_VERSION=latest SWITCH_TO_PHP=5.2
       script:
         - ./bin/run-wp-unit-tests.sh
 
-    - stage: test
+    - name: E2E tests (Admin with plugins) (1/2)
       env: WP_VERSION=latest POPULAR_PLUGINS=true
+      install:
+        - ./bin/setup-local-env.sh
+      script:
+        - $( npm bin )/jest --config test/e2e/jest.config.json --listTests > ~/.jest-e2e-tests
+        - npm run test-e2e -- --ci --runInBand --cacheDirectory="$HOME/.jest-cache" --runTestsByPath $( awk 'NR % 2 == 0' < ~/.jest-e2e-tests )
+
+    - name: E2E tests (Admin with plugins) (2/2)
+      env: WP_VERSION=latest POPULAR_PLUGINS=true
+      install:
+        - ./bin/setup-local-env.sh
+      script:
+        - $( npm bin )/jest --config test/e2e/jest.config.json --listTests > ~/.jest-e2e-tests
```

Comment:
It looks like this logic to running only the subset of test files is something that could be included in `wp-scripts test-e2e` by design with a new CLI flag:
https://github.com/WordPress/gutenberg/blob/master/packages/scripts/scripts/test-e2e.js#L45

Reply:
👍 ab1faa1d7

Reply:
We aren't able to use `npm run test-e2e` here because it uses `concurrently` which writes a bunch of extra logging to `STDOUT`. We want _only_ the test path names to be written to `STDOUT` so that `~/.jest-e2e-tests` contains only test path names. This lets us pipe it into the following command as an argument.